### ISSUE TITLE
py_to_slice() lifetime desynchronized from PyBuffer: UAF risk with mutable buffers + py.detach()

### ISSUE THREAD

**Issue by ReplikanteK** [Chia-Network/chia_rs#1488]
## Summary

`py_to_slice()` (wheel/src/run_generator.rs:20-23) returns a `&'a [u8]` slice whose lifetime `'a` is not tied to the `PyBuffer<u8>` it was derived from. Combined with `py.detach()` GIL release in 6 callers, this creates a theoretical use-after-free window if a **mutable** buffer-protocol object (e.g., `bytearray`) is passed.

## The Lifetime Desync

```rust
pub fn py_to_slice<'a>(buf: PyBuffer<u8>) -> &'a [u8] {
    assert!(buf.is_c_contiguous(), "buffer must be contiguous");
    unsafe { std::slice::from_raw_parts(buf.buf_ptr() as *const u8, buf.len_bytes()) }
}
```

`buf` is taken by value → dropped at function exit → `PyBuffer_Release` → `ob_exports--` on the underlying Python object. The returned `&'a [u8]` outlives the `PyBuffer` that held the export lock.

## The py.detach() Pattern

Affected callers (all follow the same pattern):

| Function | File | `py_to_slice` at | `py.detach()` at |
|---|---|---|---|
| `run_block_generator` | `run_generator.rs` | 53 | 55 |
| `run_block_generator2` | `run_generator.rs` | 104 | 106 |
| `additions_and_removals` | `run_generator.rs` | 146 | 148 |
| `generator_interned_vbytes` | `run_generator.rs` | 161 | 162 |
| `get_spends_for_trusted_block` | `api.rs` | 568 | 573 |
| `get_spends_for_trusted_block_with_conditions` | `api.rs` | 595 | 599 |

```rust
let program = py_to_slice::<'a>(program);  // PyBuffer dropped → ob_exports--
py.detach(|| {
    // GIL released, &[u8] points to unprotected memory
    native_fn(program, ...)
});
```

## Exploitation Chain (Theoretical)

1. Pass a `bytearray` (mutable) to the PyO3 function
2. `PyObject_GetBuffer` → `ob_exports++`
3. `py_to_slice` drops `PyBuffer` → `PyBuffer_Release` → `ob_exports--`
4. `&[u8]` points to memory now unprotected by the export lock
5. `py.detach()` releases the GIL
6. Another thread modifies the `bytearray` (e.g., `.extend()`) → **realloc** of backing buffer
7. Rust reads from the dangling `&[u8]` → **Use-After-Free**

## Current Impact

**Low.** In production, all callers in chia-blockchain pass `bytes(...)` (immutable, cannot realloc). The vulnerability requires the attacker to already have arbitrary Python code execution, which makes it redundant.

## Suggested Fix

Either:
- Tie the `&[u8]` lifetime to the `PyBuffer` (prevent the slice from outliving the export), or
- Copy the bytes out of the buffer before dropping the `PyBuffer`, or
- Move `py.detach()` before `py_to_slice()` so the GIL is already released when the buffer is created (conceptual constraint, may not always apply)

The simplest fix: replace the ownership-transfer pattern with a borrow:

```rust
pub fn py_to_slice<'a>(buf: &'a PyBuffer<u8>) -> &'a [u8] {
    assert!(buf.is_c_contiguous(), "buffer must be contiguous");
    unsafe { std::slice::from_raw_parts(buf.buf_ptr() as *const u8, buf.len_bytes()) }
}
```

This would force callers to keep the `PyBuffer` alive for the slice's duration, preventing the desync.
